src/charm.py

```diff
@@ -46,13 +46,20 @@ def _on_install(self, event: InstallEvent) -> None:
         Returns:
             None
         """
+        if self._is_magmad_enabled:
+            return
         self.unit.status = MaintenanceStatus("Installing AGW Snap")
         self.install_magma_access_gateway_snap()
         if not self._is_configuration_valid:
             self.unit.status = BlockedStatus("Configuration is invalid. Check logs for details")
             return
         self.unit.status = MaintenanceStatus("Installing AGW")
-        self.install_magma_access_gateway()
+        returncode = self.install_magma_access_gateway()
+        if returncode != 0:
+            self.unit.status = BlockedStatus("Installation script failed. See logs for details")
+            return
+        self.unit.status = MaintenanceStatus("Rebooting to apply changes")
+        self.reboot()
```

Comment:
What is going to be the status of the charm after the reboot? I'm thinking about the case when system starts and `magma` is not running yet (or it fails to start). In that case `_on_install` will exit immediately and `_on_start` will be deferred, but neither of them will set any status. Perhaps adding `WaitingStatus` before deferring `_on_start` would make sense. WDYT?

Reply:
It will stay in `MaintenanceStatus` until it has finished booting completely. The `_on_start` event will be deferred until `magma` is running.

Reply:
Cool, thanks!